ambry-replication/src/main/java/com/github/ambry/replication/RemoteReplicaInfo.java

```diff
@@ -86,6 +90,50 @@ public RemoteReplicaInfo(ReplicaId replicaId, ReplicaId localReplicaId, Store lo
     initializeTokens(token);
     // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
     this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
+    this.replicationRetryCount = 0;
+    this.MAX_REPLICATION_RETRY_COUNT = 0;
+  }
+
+  public RemoteReplicaInfo(ReplicationConfig replicationConfig, ReplicaId replicaId, ReplicaId localReplicaId,
+      Store localStore, FindToken token, long tokenPersistIntervalInMs, Time time, Port port) {
+    this.replicaId = replicaId;
+    this.localReplicaId = localReplicaId;
+    this.totalBytesReadFromLocalStore = 0;
+    this.localStore = localStore;
+    this.time = time;
+    this.port = port;
+    this.tokenPersistIntervalInMs = tokenPersistIntervalInMs;
+    initializeTokens(token);
+    // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
+    this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
+    this.replicationRetryCount = 0;
+    this.MAX_REPLICATION_RETRY_COUNT = replicationConfig.maxReplicationRetryCount;
+  }
+
+  /**
+   * Increments replicationRetryCount by 1
+   * @return replicationRetryCount
+   */
+  public long incReplicationRetryCount() {
+    this.replicationRetryCount += 1;
+    return this.replicationRetryCount;
+  }
+
+  /**
+   * Resets replicationRetryCount to 0
+   * @return replicationRetryCount
+   */
+  public long resetReplicationRetryCount() {
+    this.replicationRetryCount = 0;
+    return this.replicationRetryCount;
+  }
+
+  /**
+   * Checks if replicationRetryCount has exceeded the max number of retries allowed
+   * @return True if replicationRetryCount has exceeded the max retries, false otherwise
+   */
+  public boolean isReplicationRetryCountMaxed() {
+    return this.replicationRetryCount >= MAX_REPLICATION_RETRY_COUNT;
```

Comment:
nit: this is redundant.

Reply:
I don't understand why this is redundant. I use this method in the replicaThread.